pkg/controller/argocd/secret.go

```diff
@@ -379,6 +386,50 @@ func (r *ReconcileArgoCD) reconcileGrafanaSecret(cr *argoprojv1a1.ArgoCD) error
 	return r.client.Create(context.TODO(), secret)
 }
 
+// reconcileClusterPermissionsSecret ensures ArgoCD instance is namespace-scoped
+func (r *ReconcileArgoCD) reconcileClusterPermissionsSecret(cr *argoprojv1a1.ArgoCD) error {
+	secret := argoutil.NewSecretWithSuffix(cr.ObjectMeta, "default-cluster-config")
+	secret.Labels[common.ArgoCDSecretTypeLabel] = "cluster"
+	dataBytes, _ := json.Marshal(map[string]interface{}{
+		"tlsClientConfig": map[string]interface{}{
+			"insecure": false,
+		},
+	})
+
+	secret.Data = map[string][]byte{
+		"config":     dataBytes,
+		"name":       []byte("in-cluster"),
+		"server":     []byte(common.ArgoCDDefaultServer),
+		"namespaces": []byte(cr.Namespace),
+	}
+
+	applyReconcilerHook(cr, secret, "")
+
+	clusterSecrets := &corev1.SecretList{}
+	opts := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			common.ArgoCDSecretTypeLabel: "cluster",
+		}),
+		Namespace: cr.Namespace,
+	}
+
+	if err := r.client.List(context.TODO(), clusterSecrets, opts); err != nil {
+		return err
+	}
+
+	secrets := clusterSecrets.Items
+
+	for _, s := range secrets {
+		// check if cluster secret with name in-cluster exists
+		// do nothing if exists.
+		if string(s.Data["name"]) == "in-cluster" {
+			return nil
```

Comment:
If in-cluster secret exists, we are not updating the "namespaces" field.  I thought we needed to reconcile the namespaces field to add argocd's namespace.

Reply:
we don't want to do that right now, since we want the users to be able to manually update the Secret if they wish to grant additional permissions to the ArgoCD instance. 
So for that, i have added a `NOTE` in the description.

Reply:
This patch is supposed to fix existing argocd instances.   We should not require users to manually edit the secret the exising argoco instances.   BTW, please make sure you test the upgrade path.  If the namespaces is empty, it is likely that the users have not edit the secret.  In this case, we should add the namespace there.  Otherwise, if the namespaces is not empty, we can leave it.

Reply:
@jannfis 
I created an namespace scoped argocd instance using v1.1.1.   I went to setup and view my in-cluster cluster and did not change anything.   I then upgraded my operator.   My argocd instance is still nu-usuable.   One of the reasons is the namespaces is not patched.

Reply:
Retested successfully with latest commit.